pinot-spi/src/main/java/org/apache/pinot/spi/config/table/UpsertConfig.java

```diff
@@ -41,21 +40,21 @@ public enum Strategy {
   }
 
   @JsonPropertyDescription("Upsert mode.")
-  private final Mode _mode;
+  private Mode _mode;
```

Comment:
Consider making `_mode` default to `FULL` and `_hashFunction` default to `NONE`, which is the most common setup and these 2 fields cannot be `null`

Reply:
Thanks @Jackie-Jiang 
For `_mode` the value can be `NONE` for non-upsert tables. 
Is there any side effect on changing the default value to "FULL"?